README.md

```diff
@@ -46,3 +46,172 @@ note: the UI currently uses code directly from github, pointing to the ts-client
 
 - use `yarn publish` to release a new package, ensure compatibility with program release to mainnet-beta
 - fix the tag auto added by yarn to match our internal convention, see script `fix-npm-tag.sh`, tags should look like this e.g.`npm-v0.0.1`, note: the npm package version/tag should not necessarily match the latest program deployment
+
+
+### FAQ (on-chain program)
+
+#### Not enough CU
+*(CU: Compute Unit)*
+
+Although you want to keep an eye on the CU limit to make sure the program does not consume to many CUs, program integration test will fail unexpectedly if limit is too low when developing.
+
+Use this method to change the CU limit:
+```rust
+test.set_compute_max_units(500_000);
+```
+
+
+#### Wrong Rust version
+
+
+| Rust Version                        | Compatibility | Solana Version |
+|-------------------------------------|---------------|---------------:|
+| 1.69-x86_64-apple-darwin            | works         |         1.16.x |
+| 1.66.1-x86_64-apple-darwin          | works         |         1.14.x |
+| stable-x86_64-apple-darwin (1.69.0) | broken        |         1.14.x |
+| 1.68.2-x86_64-apple-darwin          | broken        |         1.14.x |
+| 1.67.1-x86_64-apple-darwin          | broken        |         1.14.x |
+
+Check/set the version:
+```
+rustup override list
+# WRONG:
+# /Users/username/code/mango-v4               1.69-aarch64-apple-darwin
+
+rustup override set 1.69-x86_64
```

Comment:
imo people shouldn't use `rustup override`, we already specify the version in "rust-toolchain.toml" and cargo/rustup respect that?

Reply:
referring to `rustup show` now